runtime-components/non-repudiation/src/test/scala/com/daml/nonrepudiation/NonRepudiationProxySpec.scala

```diff
@@ -138,21 +143,36 @@ final class NonRepudiationProxySpec
 object NonRepudiationProxySpec {
 
   final case class Setup[Key](
-      keys: KeyRepository,
+      certificates: CertificateRepository,
       signedPayloads: SignedPayloadRepository[Key],
+      privateKey: PrivateKey,
+      certificate: X509Certificate,
       proxyBuilder: InProcessServerBuilder,
       proxyChannel: Channel,
   )
 
   object Setup {
 
+    def generateKeyAndCertificate(): (PrivateKey, X509Certificate) = {
+      val generator = new CertAndKeyGen(AlgorithmString.RSA, AlgorithmString.SHA256withRSA)
+      generator.generate(2048)
+      val privateKey = generator.getPrivateKey
+      val certificate = generator.getSelfCertificate(
+        new X500Name("CN=Non-Repudiation Test,O=Digital Asset,L=Zurich,C=CH"),
```

Comment:
Hm, I wonder if the fact that we use CNs but don’t validate them (or really subject alt names because nobody uses CNs these days) will cause trouble. I don’t really want to get into the business of doing that so maybe just needs to be very clear in the docs that we do not validate this.

Reply:
Ultimately should we not simply trust the certificate authority (which is indeed not yet in, but will ultimately be added)?

Reply:
The certificate authority validates that it’s a valid certificate. Subject alt names validate that it’s a certificate for the correct IP/hostname. You can’t really replace one by the other afaict.

Reply:
I see, thanks for the clarification. Should we add this validation at all? If yes, I can create a ticket for this to make sure we don't lose this.

Reply:
I’d probably not add it so the task here is only to document that we do not validate this.

Reply:
Tracked by https://github.com/digital-asset/daml/issues/8746